### PR TITLE
KMS: Smithy-generated code

### PR DESCRIPTION
**Pull request by lucasmcdonald3**
_Issue #, if available:_

_Description of changes:_

_Squash/merge commit message, if applicable:_

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ComAmazonawsKms/runtimes/python/src/com_amazonaws_kms/smithygenerated/com_amazonaws_kms/aws_sdk_to_dafny.py [hunk 1]
```diff
@@ -0,0 +1,2562 @@
...
+    elif input == "SYMMETRIC_DEFAULT":
+        return KeySpec_SYMMETRIC__DEFAULT()
+    else:
+        raise ValueError("No recognized enum value in enum type: " + input)
+
+
+def com_amazonaws_kms_OriginType(input):
+    # Convert OriginType
+    if input == "AWS_KMS":
+        return OriginType_AWS__KMS()
```
Comment (ShubhamChaturvedi7) [inline ComAmazonawsKms/runtimes/python/src/com_amazonaws_kms/smithygenerated/com_amazonaws_kms/aws_sdk_to_dafny.py:723]: What's the reason of doing this conversion?

File: ComAmazonawsKms/runtimes/python/src/com_amazonaws_kms/smithygenerated/com_amazonaws_kms/dafny_to_aws_sdk.py [hunk 2]
```diff
@@ -0,0 +1,2194 @@
...
+    SigningAlgorithmSpec_RSASSA__PSS__SHA__384,
+    SigningAlgorithmSpec_RSASSA__PSS__SHA__512,
+    WrappingKeySpec_RSA__2048,
+)
+
+
+def com_amazonaws_kms_AlreadyExistsException(input):
+    output = {}
+    if input.message.is_Some:
+        output["message"] = input.message.value.VerbatimString(False)
```
Comment (ShubhamChaturvedi7) [inline ComAmazonawsKms/runtimes/python/src/com_amazonaws_kms/smithygenerated/com_amazonaws_kms/dafny_to_aws_sdk.py:102]: What does `input.KeyId.VerbatimString(False)` do?

**Review by ShubhamChaturvedi7 — COMMENTED** [review]
Reviewed some random operation implementations since the TypeConversion layer is almost 10k lines. They all should follow the same pattern, but it's better to get a second pair of eyes.

I assume CI is failing because not all the code pieces are checked in right now?

File: ComAmazonawsKms/runtimes/python/src/com_amazonaws_kms/smithygenerated/com_amazonaws_kms/dafny_to_aws_sdk.py (hunk 2 above)
Comment (lucasmcdonald3) [reply ComAmazonawsKms/runtimes/python/src/com_amazonaws_kms/smithygenerated/com_amazonaws_kms/dafny_to_aws_sdk.py:102]: This converts a Dafny key ID to a native key ID
Dafny key ID is a `_dafny.Seq`. Seq has a method VerbatimString: [link](https://github.com/dafny-lang/dafny/blob/385103dd9e77a329b04c45fef0c3ceee1ae6bb26/Source/DafnyRuntime/DafnyRuntimePython/_dafny.py#L155)
Dafny strings are represented as tuples of characters in Python.
VerbatimString(False) will `''.join` the Dafny characters together.
This gets you to a native string

File: ComAmazonawsKms/runtimes/python/src/com_amazonaws_kms/smithygenerated/com_amazonaws_kms/aws_sdk_to_dafny.py (hunk 1 above)
Comment (lucasmcdonald3) [reply ComAmazonawsKms/runtimes/python/src/com_amazonaws_kms/smithygenerated/com_amazonaws_kms/aws_sdk_to_dafny.py:723]: This gets you from the native enum `OriginType` to its Dafny representation
I found this API page which seems to line up with this API: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/kms/client/describe_key.html
Interestingly, we appear to be missing `EXTERNAL_KEY_STORE` from our model... https://github.com/aws/aws-cryptographic-material-providers-library/blob/main/ComAmazonawsKms/Model/kms/model.json#L3919-L3937

Comment (lucasmcdonald3) [reply ComAmazonawsKms/runtimes/python/src/com_amazonaws_kms/smithygenerated/com_amazonaws_kms/aws_sdk_to_dafny.py:723]: https://sim.amazon.com/issues/CrypTool-5286 for bookkeeping on this

**Comment by lucasmcdonald3** [comment]
> Reviewed some random operation implementations since the TypeConversion layer is almost 10k lines. They all should follow the same pattern, but it's better to get a second pair of eyes.

Yep, I think this is the right way to go about this

> I assume CI is failing because not all the code pieces are checked in right now?

Correct -- for any PRs until all code is checked in, refer to the CI from https://github.com/aws/aws-cryptographic-material-providers-library/pull/171

**Review by justplaz — APPROVED** [review]
LGTM.

**Review by ShubhamChaturvedi7 — APPROVED** [review]

**Pull request merged**